pyramid_learning_journal/pyramid_learning_journal/tests.py

```diff
@@ -1,78 +1,83 @@
-#     assert response.status_code == 200
+from pyramid_learning_journal.models.meta import Base
 
 # =================== FUNCTIONAL TESTS =============
 
 
 @pytest.fixture
-def testapp():
+def testapp(request):
     """Create a test application to use for functional tests."""
     from pyramid_learning_journal import main
     from webtest import TestApp
+    from pyramid.config import Configurator
+
+    def main(global_config, **settings):
+        """ This function returns a Pyramid WSGI application.
+        """
+        settings['sqlalchemy.url'] = os.environ.get('DATABASE_URL_TEST')
+        config = Configurator(settings=settings)
+        config.include('pyramid_jinja2')
+        config.include('.models')
+        config.include('.routes')
+        config.scan()
+        return config.make_wsgi_app()
+
     app = main({})
-    return TestApp(app)
+    testapp = TestApp(app)
+
+    SessionFactory = app.registry["dbsession_factory"]
+    engine = SessionFactory().bind
+    Base.metadata.create_all(bind=engine)
+
+    def tearDown():
+        Base.metadata.drop_all(bind=engine)
+
+    request.addfinalizer(tearDown)
+
+    return testapp
 
 
 def test_home_route_returns_home_content(testapp):
     """Test the thome route returns home content."""
     response = testapp.get('/')
     html = response.html
     assert 'Journal' in str(html.find('h1').text)
-    assert 'David' in str(html.find('title').text)
+    assert 'Journal' in str(html.find('title').text)
 
 
-def test_home_route_listing_has_all_journals(testapp):
-    """Test the home route listing has all journals."""
-    response = testapp.get('/')
+def test_home_route_has_h1(testapp):
+    """The home page has a table in the html."""
+    response = testapp.get('/', status=200)
     html = response.html
-    assert len(JOURNALS) == len(html.find_all('li'))
+    assert len(html.find_all("h1")) == 2
+
+
+# def test_detail_view_returns_dict(testapp):
+#     """Home view returns a dictionary of values."""
+#     from pyramid_learning_journal.views.default import detail_view
+#     response = detail_view(testapp)
+#     assert isinstance(response, dict)
+
+
+# def test_detail_view_returns_count_matching_database(testapp):
+#     """Home view response matches database count."""
+#     from pyramid_learning_journal.views.default import detail_view
+#     response = detail_view(testapp)
+#     query = testapp.dbsession.query(Journal)
+#     assert len(response['journal']) == query.count()
+
+
+# def test_detail_route_returns_proper_content(testapp):
+#     """Test the detail route returns proper journal entry."""
+#     response = testapp.get('/journal/1')
+#     html = response.html
+#     import pdb; pdb.set_trace()
+#     assert
 
 
-def test_detail_route_with_bad_id(testapp):
-    """Test the detail route with a bad ID."""
-    response = testapp.get('/journal/400', status=404)
-    assert "Alchemy scaffold" in response.text
+    # config.add_route('detail', '/journal/{id:\d+}')
```

Comment:
Your tests error out. Get the tests working and add a lot more. I want to see tests for expected function, expected failure, and edge cases.